site/cssGen.js

```diff
@@ -0,0 +1,51 @@
+const fs = require("fs");
+const path = require("path");
+
+function getCssVariablesFromDir(dirPath) {
+  const cssVariableRegex = /([a-zA-Z0-9_-]+)\s*:\s*([^;]+)/g;
+  const cssVariables = {};
+
+  const files = fs.readdirSync(dirPath);
+
+  files.forEach((file) => {
+    const filePath = path.join(dirPath, file);
+    const stats = fs.statSync(filePath);
+
+    if (stats.isDirectory()) {
+      // Recursively process subdirectories
+      Object.assign(cssVariables, getCssVariablesFromDir(filePath));
+    } else if (stats.isFile() && path.extname(file) === ".css") {
+      // Process CSS files
+      const cssContent = fs.readFileSync(filePath, "utf8");
+      let match;
+
+      while ((match = cssVariableRegex.exec(cssContent)) !== null) {
+        const variableName = match[1];
+        const variableValue = match[2].trim();
+        cssVariables[variableName] = variableValue;
+      }
+    }
+  });
+
+  return cssVariables;
+}
+
+const themeDirPath = path.resolve(
+  __dirname,
+  "../packages/theme/css/characteristics"
+);
+
+const allCssVariables = getCssVariablesFromDir(themeDirPath);
+const jsonData = JSON.stringify(allCssVariables, null, 2);
+const cssFolderPath = path.resolve(
+  __dirname,
+  "../site/src/components/css-display"
+);
+
+const outputPath = path.join(cssFolderPath, "cssVariables.json");
+
+try {
+  fs.writeFileSync(outputPath, jsonData, "utf8");
```

Comment:
Should the output file be added to `.gitignore`?

Reply:
i dont think it should matter, because it will overwrite the file if it exists or generate one if it doesn't

Reply:
The file is not *source code*, and the file committed may not reflect the latest information needed. Being source controlled may be misleading